src/pvdr/kprobe.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015-2017 Tobias Waldekranz <tobias@waldekranz.com>
+ *
+ * This file is part of ply.
+ *
+ * ply is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, under the terms of version 2 of the
+ * License.
+ *
+ * ply is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ply.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _KPROBE_H
+#define _KPROBE_H
+
+#define KPROBE_MAXLEN   0x100
+
+/*
+ * Structure used for internal representation of kprobes, kretprobes and
+ * tracepoints.
+ */
+typedef struct kprobe {
+	const char *pvdr;
+	const char *type;
+	FILE *ctrl;
+	int bfd;
+
+	struct {
+		int cap, len;
+		int *fds;
+	} efds;
+} kprobe_t;
+
+extern const module_t *kprobe_modules[];
+extern const module_t *kretprobe_modules[];
+
```

Comment:
Many of the following functions are not used by other providers so don't expose them in the header.

If there are ordering issues, please put the required prototypes in `kprobe.c`.